File: WPF/Controls/PopupShelter.vb
Namespace Common.Controls

    Public Class PopupShelter
        Inherits Control

        Shared Sub New()
            DefaultStyleKeyProperty.OverrideMetadata(GetType(PopupShelter), New FrameworkPropertyMetadata(GetType(PopupShelter)))
        End Sub

#Region "IsShelterShown Property"
        Private Shared ReadOnly IsShelterShownPropertyKey As DependencyPropertyKey = DependencyProperty.RegisterReadOnly("IsShelterShown", GetType(Boolean), GetType(PopupShelter), New PropertyMetadata(True))
        Public Shared ReadOnly IsShelterShownProperty As DependencyProperty = IsShelterShownPropertyKey.DependencyProperty

        Public Property IsShelterShown As Boolean
            Get
                Return DirectCast(Me.GetValue(IsShelterShownProperty), Boolean)
            End Get
            Friend Set(ByVal value As Boolean)
                Me.SetValue(IsShelterShownPropertyKey, value)
            End Set
        End Property
#End Region

    End Class

End Namespace

File: WPF/Controls/PopupShelter.cs
using System.Windows;
using System.Windows.Controls;

namespace Ks.Common.Controls
{
    public class PopupShelter : Control
    {
        static PopupShelter()
        {
            DefaultStyleKeyProperty.OverrideMetadata(typeof(PopupShelter), new FrameworkPropertyMetadata(typeof(PopupShelter)));
        }

        private static readonly DependencyPropertyKey IsShelterShownPropertyKey = DependencyProperty.RegisterReadOnly("IsShelterShown", typeof(bool), typeof(PopupShelter), new PropertyMetadata(true));
        public static readonly DependencyProperty IsShelterShownProperty = IsShelterShownPropertyKey.DependencyProperty;

        public bool IsShelterShown
        {
            get => (bool) this.GetValue(IsShelterShownProperty);
            internal set => this.SetValue(IsShelterShownPropertyKey, value);
        }
    }
}
